### PR TITLE
Add a basic DNSConfig class

### PR DESCRIPTION
**Pull request by imobachgs**
This PR adds:

* a Y2Network::DNS class
* a reader for DNS configuration
* a presenter for the DNS summary

### CODE REVIEW

File: src/lib/y2network/config_reader/sysconfig_dns.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "yast"
+require "y2network/dns_config"
+
+module Y2Network
+  module ConfigReader
+    # Reads DNS configuration from sysconfig files
+    class SysconfigDNS
```
Comment (jreidinger) [inline src/lib/y2network/config_reader/sysconfig_dns.rb]: Hmm, to be honest, I do not like much this double name. For me it a bit indicate that maybe we need namespace for ConfigReader::Syconfig and there having stuff like DNS. What do you think?

Comment (imobachgs) [reply src/lib/y2network/config_reader/sysconfig_dns.rb]: Yes, I prefer that approach too. But I have found sometimes some resistance to have more than 2 levels :) If you are fine, I will move it as I prefer that approach too.

Comment (jreidinger) [reply src/lib/y2network/config_reader/sysconfig_dns.rb]: well, I am not fan of more then 2 levels, but in this case it makes some sense to me. So as with any rule, use common sense :) Having three levels for few classes is too much for me, but having it for set of class as separator if there are more of such classes, makes sense for me. Another approach can be `Y2Network::ConfigReader::DNS::Sysconfig` where you basically need to use only `ConfigReader::DNS.from_system` to read what is currently used.

File: src/lib/y2network/dns_config.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Network
+  # DNS configuration (hostname, name servers, etc.).
+  class DNSConfig
```
Comment (jreidinger) [inline src/lib/y2network/dns_config.rb]: well, when I see all that separation. Maybe it makes more sense to have `Y2Network::DNS` or `Y2DNS` and inside config reader, config and others? Still thinking about reasonable naming and structure to be

1. intuitive
2. easy to read
3. easy to write
4. as short as possible while still above is valid.

Comment (teclator) [reply src/lib/y2network/dns_config.rb]: I also had the same doubts, and liked Y2Routing, Y2DNS but it was argued that they where below y2network namespace, so Y2Network::Routing && Y2Network::DNS looks reasonable to me.

File: src/lib/y2network/config_reader/sysconfig_dns.rb [hunk 3]
```diff
@@ -74,6 +79,39 @@ def resolv_conf_policy
       #
       # @return [String]
       def hostname
```
Comment (teclator) [inline src/lib/y2network/config_reader/sysconfig_dns.rb]: In yast/yast-yast2#921 I added a `Hostname` class and a reader only for the `Hostname`, just for cases where the complete `DNS` config is not needed at all but only the `hostname`. 

It is committed as [RFC].

File: src/lib/y2network/config.rb [hunk 4]
```diff
@@ -83,10 +86,12 @@ def configs
     #
     # @param interfaces [Array<Interface>] List of interfaces
     # @param routing    [Routing] Object with routing configuration
+    # @param dns        [DNS] Object with DNS configuration
```
Comment (jreidinger) [inline src/lib/y2network/config.rb]: well, nil is not documented. So I suggest to simply use as default value Y2Network::DNS.new and not have nil anywhere.

**Pull request merged**